### PR TITLE
feat: add Windows CI

### PR DESCRIPTION
**Pull request by Samarium150**
- [ ] Implementation
- [x] Windows CI

~~Windows CI may cause a lot of trouble because of Qt.~~
Windows CI does cause a lot of trouble, but because of compiling errors.

### CODE REVIEW

**Comment by Samarium150** [comment]
OK, Windows CI is fully functional

File: test/solver_test.cpp
Comment (pwyq) [inline test/solver_test.cpp:1]: all other looks good so far!

but are you sure about the changes here

File: test/solver_test.cpp [hunk 1]
```diff
@@ -297,10 +297,10 @@ TEST(LARGE, 4) {
   EXPECT_EQ(dfpn.formatResult().substr(0, 7), "W 7 9 3");
 }
 
-}  // namespace dfpn
```
Comment (pwyq) [inline test/solver_test.cpp:300]: all other looks good so far!

but are you sure about the changes here

Comment (Samarium150) [reply test/solver_test.cpp:300]: This one looks a little bit weird. I'll fix it in the next commit.

**Pull request merged**